pym/bob/invoker.py

```diff
@@ -305,6 +307,12 @@ async def executeStep(self, mode, clean=False, keepSandbox=False):
             elif clean and mode != InvocationMode.SHELL:
                 emptyDirectory(self.__spec.workspaceWorkspacePath)
 
+            if len(self.__makeFds) == 2:
+                self.__spec.env["MAKEFLAGS"] = ("".join([self.__spec.env.get("MAKEFLAGS") if
+                        self.__spec.env.get("MAKEFLAGS") is not None else ""])
+                    + " -j" + str(self.__makeJobs) + " --jobserver-auth="
+                    + ",".join([str(fd) for fd in self.__makeFds]))
```

Comment:
Hmm, I've played a bit around and it looks like `make` will parse the `MAKEFLAGS` that it gets, forwards some of the options it knows and overwrites the rest. Maybe we should do the same?

Reply:
I don't think we should remove or edit the `MAKEFLAGS` defined in the recipe or somewhere else? What sounds reasonable is to overwrite `-j` and `--jobserver-auth` if they are already in the `MAKEFLAGS`?

Reply:
I'm not sure yet how to handle it properly. By default I would probably only forward options that are understood by Bob. For example if you define variables on the command line, e.g. `make -j3 FOO=bar`, you will get `MAKEFLAGS= -j3 --jobserver-auth=3,4 -- FOO=bar`. That will influence the build and we should probably strip the assignment out.

OTOH you are probably right that we should not overwrite what a user did specify in the recipe or on the command line.

So maybe we should just use `-j` and `--jobserver-auth` from the OS environment. But if the user specified something in the recipes then we amend `MAKEFLAGS` and keep the rest...